lib/trento.pm

```diff
@@ -220,13 +224,42 @@ sub config_cluster {
         get_var('QESAPDEPLOY_IMDB_CLIENT'));
 }
 
+=head3 deploy_vm
+
+Deploy the main VM for the Trento application
+Based on 00.040-trento_vm_server_deploy_azure.sh
+=cut
+
+sub deploy_vm {
+    # Run the Trento deployment
+
+    my $script_id = '00.040';
+    my $resource_group = get_resource_group();
+    my $machine_name = get_vm_name();
+    record_info($script_id);
+    my $vm_image = get_var(TRENTO_VM_IMAGE => 'SUSE:sles-sap-15-sp3-byos:gen2:latest');
+    my $deploy_script_log = "script_$script_id.log.txt";
+    my $cmd = join(' ', TRENTO_SCRIPT_RUN . $script_id . '-trento_vm_server_deploy_azure.sh ',
+        '-g', $resource_group,
+        '-s', $machine_name,
+        '-i', $vm_image,
+        '-a', VM_USER,
+        '-k', SSH_KEY . '.pub',
+        '-v', "2>&1|tee $deploy_script_log");
+    assert_script_run($cmd, 360);
+    upload_logs($deploy_script_log);
+}
+
 =head3 deploy_qesap
 
 Deploy a SAP Landscape using a previously configured qe-sap-deployment
 =cut
 
 sub deploy_qesap {
     qesap_sh_deploy(SSH_KEY);
+    my $inventory = qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER'));
```

Comment:
Why you replace get_required_var('PUBLIC_CLOUD_PROVIDER') for argument in lib/qesapdeployment.pm and then use get_required_var('PUBLIC_CLOUD_PROVIDER') here ?
Also the argument is not used anywhere in this PR, it's just confusing for me.

Reply:
I like the idea to not use any PUBLIC_CLOUD variables directly in the qesapdeploy library, as the qesapdeploy library is not part of any publiccloud library. For this reason, I moved one layer up this dependency. 
trento is not part of PC too so the same is valid for trento lib too, but I decided to move one step at the time.
The next step is to move get_required_var('PUBLIC_CLOUD_PROVIDER') one layer up, in the test